frontend/test/metabase/scenarios/admin/datamodel/table.cy.spec.js

```diff
@@ -53,4 +56,58 @@ describe("scenarios > admin > databases > table", () => {
       cy.findByText("Creation timestamp");
     });
   });
+
+  describe.skip("turning table visibility off shouldn't prevent editing related question (metabase#15947)", () => {
+    it("simple question (metabase#15947-1)", () => {
+      turnTableVisibilityOff(ORDERS_ID);
+      cy.visit("/question/1");
+      cy.findByText("Filter");
+    });
+
+    it("question with joins (metabase#15947-2)", () => {
+      cy.createQuestion({
+        name: "15947",
+        query: {
+          "source-table": ORDERS_ID,
+          joins: [
+            {
+              fields: "all",
+              "source-table": PRODUCTS_ID,
+              condition: [
+                "=",
+                ["field", ORDERS.PRODUCT_ID, null],
+                ["field", PRODUCTS.ID, { "join-alias": "Products" }],
+              ],
+              alias: "Products",
+            },
+          ],
+          filter: [
+            "and",
+            ["=", ["field", ORDERS.QUANTITY, null], 1],
+            [">", ["field", PRODUCTS.RATING, { "join-alias": "Products" }], 3],
+          ],
+          aggregation: [
+            ["sum", ["field", ORDERS.TOTAL, null]],
+            ["sum", ["field", PRODUCTS.RATING, { "join-alias": "Products" }]],
+          ],
+          breakout: [
+            ["field", ORDERS.CREATED_AT, { "temporal-unit": "year" }],
+            ["field", PRODUCTS.CATEGORY, { "join-alias": "Products" }],
+          ],
+        },
+      }).then(({ body: { id: QUESTION_ID } }) => {
+        turnTableVisibilityOff(PRODUCTS_ID);
+        cy.visit(`/question/${QUESTION_ID}/notebook`);
+        cy.findByText("Quantity is equal to 1");
```

Comment:
Let's just make sure that the table is in the join too
```suggestion
        cy.findByText("Products");
        cy.findByText("Quantity is equal to 1");
```